src/Microsoft.AspNet.Mvc.TagHelpers/SelectTagHelper.cs

```diff
@@ -102,25 +83,47 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // "SelectExpressionNotEnumerable" InvalidOperationException during generation.
             // Metadata.IsEnumerableType is similar but does not take runtime type into account.
             var realModelType = For.ModelExplorer.ModelType;
-            var allowMultiple = typeof(string) != realModelType &&
+            _allowMultiple = typeof(string) != realModelType &&
                 typeof(IEnumerable).GetTypeInfo().IsAssignableFrom(realModelType.GetTypeInfo());
+            _currentValues = Generator.GetCurrentValues(
+                ViewContext,
+                For.ModelExplorer,
+                expression: For.Name,
+                allowMultiple: _allowMultiple);
+
+            // Whether or not (not being highly unlikely) we generate anything, could update contained <option/>
+            // elements. Provide selected values for <option/> tag helpers. They'll run next.
```

Comment:
Remove `They'll run next.` since it's no longer relevant exactly when any child tag helpers execute.